### PR TITLE
Disabling msbuild node reuse for unix.

### PR DESCRIPTION
**Pull request by livarcocc**
This is to improve our CI build perf due to https://github.com/dotnet/corefx/issues/28791

### CODE REVIEW

File: run-build.sh [hunk 1]
```diff
@@ -151,6 +151,9 @@ export VSTEST_TRACE_BUILD=1
 # Don't resolve shared frameworks from user or global locations
 export DOTNET_MULTILEVEL_LOOKUP=0
 
+# Turn off MSBuild Node re-use on unix based buils - https://github.com/dotnet/corefx/issues/28791
```
Comment (nguerrera) [inline run-build.sh]: Typo: s/buils/builds/

**Comment by nguerrera** [comment]
> This is to improve our CI build perf due to dotnet/corefx#28791

I don't think the perf issue is related to that bug. The symptom would be an outright hang.

**Comment by nguerrera** [comment]
cc @rainersigwald

**Comment by rainersigwald** [comment]
It's possible that a perf degradation in these tests is a result of the other manifestation of https://github.com/Microsoft/msbuild/issues/3161 which causes independently-launched builds to serialize rather than just execute in parallel, but I suspect that isn't the case here. Did this reduce CI time?

**Comment by rainersigwald** [comment]
Note that if we take https://github.com/Microsoft/msbuild/pull/3162 we shouldn't do this; better to be able to turn it on with one revert.

**Pull request merged**